kafka-admin/src/main/java/io/strimzi/admin/kafka/admin/KafkaAdminService.java

```diff
@@ -38,22 +55,33 @@
 
                 final TypeDefinitionRegistry typeDefinitionRegistry = schemaParser.parse(userInputStream);
                 p.complete(typeDefinitionRegistry);
-            }
-            catch (Exception exc) {
+            } catch (Exception exc) {
                 p.fail(exc);
             }
         }, ar -> {
-            final TypeDefinitionRegistry schema = (TypeDefinitionRegistry) ar.result();
+                final TypeDefinitionRegistry schema = (TypeDefinitionRegistry) ar.result();
+
+                AdminClientWrapper acw = new AdminClientWrapper(vertx, config);
+                try {
+                    acw.open();
+                } catch (Exception e) {
+                    log.error("AdminClient with configuration {} cannot be created. Check whether the kafka cluster available.", config, e);
+                    System.exit(1);
```

Comment:
Seems odd for #getRegistrationDescriptor to call system.exit().  can't the future just be completed exceptionally and let the caller decide the appropriate response?